src/providers/EtherspotContextProvider.tsx

```diff
@@ -293,14 +295,44 @@ const EtherspotContextProvider = ({
       await Promise.all(
         supportedChains
           .filter(({ chainId }) => chainId !== CHAIN_ID.AVALANCHE)
-          .map(async (element) => {
+          .map(async (chain) => {
             try {
               let supportedAssets = await getSupportedAssetsWithBalancesForChainId(
-                  element.chainId,
+                  chain.chainId,
                   true,
                   walletAddress,
                   false
                 );
+              balanceByChain.push({
+                title: chain.title,
+                chain: chain.chainId,
+                total: sumAssetsBalanceWorth(supportedAssets),
+              });
+            } catch (e) {
+              //
+            }
+          })
+      );
+      setSmartWalletBalanceByChain(balanceByChain);
+    },
+    [sdk, accountAddress]
+  );
+
+  const getKeybasedWalletBalancesPerChain = useCallback(
+    async (walletAddress: string, supportedChains: Chain[]) => {
+      if (!sdk || !walletAddress) return;
+      let balanceByChain: IBalanceByChain[] = [];
+      await Promise.all(
+        supportedChains
+          .filter((element) => element.chainId !== CHAIN_ID.AVALANCHE)
+          .map(async (element) => {
+            try {
+              let supportedAssets = await getSupportedAssetsWithBalancesForChainId(
```

Comment:
this might cause API rate limit to block u, solutions are either put some cooldowns or make it sequential (might cause UX problems), but if you checked and it's ok then all good

Reply:
you mean some sort of throttling?